internal/impl/parquet/processor_encode.go

```diff
@@ -270,6 +270,7 @@ func (s *parquetEncodeProcessor) ProcessBatch(ctx context.Context, batch service
 	buf := bytes.NewBuffer(nil)
 	pWtr := parquet.NewGenericWriter[any](buf, s.schema, parquet.Compression(s.compressionType))
 
+	batch = batch.Copy()
 	rows := make([]any, len(batch))
 	for i, m := range batch {
 		ms, err := m.AsStructured()
```

Comment:
AsStructured
```suggestion
		ms, err := m.AsStructuredMut()
```

Reply:
Oh, OK. Guess the batch still needs to be copied as well since the test I added fails otherwise.

Reply:
Also, I had a quick look through this API and found this redundant assignment: https://github.com/redpanda-data/benthos/blob/50a5cd25a0c1dcef357a1467a0f2ddf556bee8d7/internal/message/data.go#L74 I wonder why the linter isn't catching it. Might be worth revising that as well.

Reply:
Wow good catch, I'll raise a PR for it.